src/dg/strong_dg.cpp

```diff
@@ -1719,19 +1734,23 @@ void DGStrong<dim,nstate,real,MeshType>::assemble_boundary_term_strong(
         for(int idim=0; idim<dim; idim++){
             surf_flux_node[idim] = metric_oper.flux_nodes_surf[iface][idim][iquad];
         }
-        this->pde_physics_double->boundary_face_values (boundary_id, surf_flux_node, unit_phys_normal_int, soln_state, aux_soln_state, soln_boundary, grad_soln_boundary);
+        //I am not sure if BC should be from solution interpolated to face
+        //or solution from the projected entropy variables.
+        //Now, it uses projected entropy variables for NSFR, and solution
+        //interpolated to face for conservative DG.
+        this->pde_physics_double->boundary_face_values (boundary_id, surf_flux_node, unit_phys_normal_int, soln_state_int, aux_soln_state_int, soln_boundary, grad_soln_boundary);
         
         // Convective numerical flux.
         std::array<real,nstate> conv_num_flux_dot_n_at_q;
-        conv_num_flux_dot_n_at_q = this->conv_num_flux_double->evaluate_flux(soln_state, soln_boundary, unit_phys_normal_int);
+        conv_num_flux_dot_n_at_q = this->conv_num_flux_double->evaluate_flux(soln_state_int, soln_boundary, unit_phys_normal_int);
```

Comment:
@AlexanderCicchino This is a good catch but we need to do a more thorough review (with another student) including the boundary terms since soln_boundary and the gradient still depend on _int values.

Reply:
> A good test would be checking entropy conservation with the boundary condition entropy production to machine precision.

Would the aforementionned test have caught this?

Reply:
Yes, I agree Prof. Nadarajah. 
Yes Doug, the aforementioned test would catch it, provided it checks both entropy conservation and global conservation.